lib/setupmanagers/qemuhck/dhcpd.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require 'fileutils'
+
+# AutoHCK module
+module AutoHCK
+  # Dhcpd class
+  class Dhcpd
+    def initialize(project)
+      @ip_segment = project.config['ip_segment']
+      @workspace_path = project.workspace_path
+    end
+
+    def run(dump_only: false)
+      conf_path = File.join(@workspace_path, 'dhcpd.conf')
+      leases_path = File.join(@workspace_path, 'dhcpd.leases')
+
+      dump_conf conf_path
+      FileUtils.touch leases_path
+
+      if dump_only
+        IO.write File.join(@workspace_path, 'dhcpd_manual.sh'), <<~SH
+          #!/usr/bin/env bash
+          dhcpd -f -d -cf #{conf_path} -lf #{leases_path} --no-pid
+        SH
+      else
+        @pid = spawn('dhcpd', '-f', '-d', '-cf', conf_path, '-lf', leases_path, '--no-pid')
+      end
+    end
+
+    def abort
+      return if @pid.nil?
+
+      Process.kill(:KILL, @pid)
+      Process.wait(@pid)
+    end
+
+    private
+
+    # rubocop:disable Metrics/MethodLength
+    def dump_conf(conf_path)
+      File.open(conf_path, 'w') do |f|
+        f.write <<~CONF
```

Comment:
Also, you can use ruby/erb and move this config to a separate file. Just an idea.